src/dft/src/config/DftConfig.cpp

```diff
@@ -34,33 +34,46 @@
 
 namespace dft {
 
-ScanArchitectConfig* DftConfig::getMutableScanArchitectConfig()
-{
-  return &scan_architect_config_;
-}
+namespace {
+constexpr std::string_view kDefaultTestModeName = "default";
+}  // namespace
 
-const ScanArchitectConfig& DftConfig::getScanArchitectConfig() const
+void DftConfig::report(utl::Logger* logger) const
 {
-  return scan_architect_config_;
+  logger->report("***************************");
+  logger->report("DFT Config Report:\n");
+
+  for (const auto& [test_mode, test_mode_config] : test_modes_config_) {
+    logger->report("\nTest mode: {}", test_mode);
+    test_mode_config.report(logger);
+  }
+
+  logger->report("***************************");
 }
 
-ScanStitchConfig* DftConfig::getMutableScanStitchConfig()
+TestModeConfig* DftConfig::createTestMode(const std::string& name)
 {
-  return &scan_stitch_config_;
+  auto pair = test_modes_config_.try_emplace(name, name);
+  return &pair.first->second;
 }
 
-const ScanStitchConfig& DftConfig::getScanStitchConfig() const
+TestModeConfig* DftConfig::getOrDefaultMutableTestModeConfig(
+    const std::string& name)
 {
-  return scan_stitch_config_;
+  auto found = test_modes_config_.find(name);
+  if (found == test_modes_config_.end()) {
+    if (name == kDefaultTestModeName) {
+      return createTestMode(name);
+    }
+    return nullptr;
```

Comment:
getOrDefaultMutableTestModeConfig sounds like it should return default if not found.